codalab/client/local_bundle_client.py

```diff
@@ -530,21 +511,75 @@ def select_unreadable_worksheets(uuids):
 
         return bundle_dict
 
-    # Return information about an individual target inside the bundle.
+    def check_download_permission(self, target):
```

Comment:
Why do we need a separate method?  If one can read the entire bundle iff one can read the target.

If you want a separate method, rename to check_target_has_read_permission.